packages/eslint-plugin/src/util/collectUnusedVariables.ts

```diff
@@ -439,6 +439,17 @@ function isExported(variable: TSESLint.Scope.Variable): boolean {
   });
 }
 
+const LOGICAL_ASSIGNMENT_OPERATORS = new Set(['&&=', '||=', '??=']);
+
+/**
+ * Checks whether the given operator is a logical assignment operator or not.
+ * @param operator The operator
+ * @returns True if the operator is a logical assignment operator.
+ */
+function isLogicalAssignmentOperator(operator: string): boolean {
```

Comment:
[Style] Just curious, is it a strong preference on your end to make this a function? I can see an advantage of it being a bit more natural to read in the `isUsedVariable` function, but it's a lot more lines this way.

I'm not particularly passionate about the issue, but would prefer it just be inlined (unless you have strong feelings).

Reply:
Thanks. I think the code reads well enough without the function, I'll change it.